src/main/kotlin/org/ionproject/integration/file/implementation/AbstractFileDownloader.kt

```diff
@@ -0,0 +1,47 @@
+package org.ionproject.integration.file.implementation
+
+import java.io.File
+import java.net.HttpURLConnection
+import java.net.URL
+import java.nio.file.Path
+import org.ionproject.integration.file.`interface`.FileDownloader
+import org.ionproject.integration.file.exception.ServerErrorException
+import org.ionproject.integration.utils.Try
+
+abstract class AbstractFileDownloader() :
+    FileDownloader {
+
+    override fun download(urlStr: String, localDestination: String): Try<Path> {
+        if (urlStr.isEmpty() || localDestination.isEmpty()) {
+            return Try.ofError(IllegalArgumentException("Parameters url and localDestination need not be empty"))
+        }
+
+        val url = URL(urlStr)
+
+        val httpConn = Try.of(url.openConnection() as HttpURLConnection)
```

Comment:
We should depend on JDK11 and use https://docs.oracle.com/en/java/javase/11/docs/api/java.net.http/java/net/http/HttpClient.html

Reply:
Opened issue #56  for updating to JDK11. Will be working on it! Do you think we should follow an asynchronous model for downloading the file? HttpClient supports it, although we would be working with Completable Futures: every map operation would read like someTry.map { cf -> cf.thenApply { elem -> elem.doSomething() } }. Do you know of any 'cleaner' alternative? For its purpose in the timetable job, PDFFileDownloader does not necessarily need to be assyncronous, as further processing of the document is only possible when it is saved to the file-system. On the other hand, we want to have re-usable components, so this FileDownloader must be adapted to wider-use.